src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/SpanNDHelpers.T.cs

```diff
@@ -0,0 +1,310 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
+
+
+#pragma warning disable 8500 // sizeof of managed types
+
+namespace System.Numerics.Tensors
+{
+    internal static partial class SpanNDHelpers // .T
+    {
+        public static unsafe void Memmove<T>(Span<T> destination, ReadOnlySpan<T> source, nint length, nint dstOffset = 0)
+        {
+            fixed (T* dst = destination)
+            fixed (T* src = source)
+            {
+                NativeMemory.Copy(src, dst + dstOffset, (nuint)length * (nuint)sizeof(T));
+            }
+        }
+
+        public static unsafe void Memmove<T>(ref T[] destination, ref T source, nint length)
+        {
+            fixed (T* dst = destination)
+            fixed (T* src = &source)
+            {
+                NativeMemory.Copy(src, dst, (nuint)length * (nuint)sizeof(T));
+            }
+        }
+
+        public static unsafe void Memmove<T>(ref T destination, ref T source, nint length)
+        {
+            fixed (T* dst = &destination)
+            fixed (T* src = &source)
+            {
+                NativeMemory.Copy(src, dst, (nuint)length * (nuint)sizeof(T));
+            }
+        }
+
+        public static unsafe void Fill<T>(ref T refData, nuint numElements, T value)
```

Comment:
This can just call `Span.Fill`, doing it `(numElements / int.MaxValue) + 1` times

Reply:
So I am now doing `Span.Fill`, but im just looping until all has been filled. Would it be better to switch to an explicit loop count like you have here?

Reply:
Fix so that it will work correctly with a slice that isn't contiguous.